### DISCUSSION TITLE
Help with calling a julia function from python when the input is an array of arrays

### DISCUSSION THREAD

**Discussion by mestinso** [cjdoris/PythonCall.jl#556] [Q&A] [answered]
If I have a Julia function that looks like the following:
`myfunc(x::AbstractVector{<:AbstractVector{<:Real}})`, how do I call that from python using juliacall in the simplest way possible?

I've tried `jl.myfunc(np.array([np.array([1,2,3])]))` but that was unsuccessful.

Got an error like the following: "...expected AbstractVector{<:AbstractVector{<:Real}}, got a value of type PyArray{Int64, 2, true, false, Int64}..."

@cjdoris I will note I found this [thread](https://discourse.julialang.org/t/juliacall-pass-numpy-array-to-julia-function-as-vector-float64/87728/5) and I did get your option 4 working. But I was hoping I could get something along the lines of your option 1 working. Note that for situations where my input is a vector I've been successful, where I've failed is a vector of vectors.

**Comment by cjdoris** [comment, marked as answer]
AFAIR `np.array([np.array([1,2,3])])` is actually a matrix, not a vector of vectors. There are no conversions in PythonCall from a matrix to a vector of vectors, so you'll need to make it a vector of vectors yourself.

You could just have a list `[np.array([1,2,3])]` or do `np.array([np.array([1,2,3])], dtype=object)`. However neither of these will convert to a `AbstractVector{<:AbstractVector{<:Real}}` because now the type of the elements of the given list or array cannot be inferred (the element type is just object).

I think you'll need to do some manual conversion using `juliacall.convert` or do the conversion Julia-side.

**Comment by mestinso** [comment]
Thanks. For anybody who might be reading this, the following is basically what I did:
`jl.myfunc(jl.convert(jl.Vector[jl.Vector[jl.Float64]],[[1,2,3],[20,50,90]]))`
